internal_configure.bzl

```diff
@@ -21,7 +20,8 @@ def _internal_configure_extension_impl(module_ctx):
         name = "pybind11",
         build_file = "//:pybind11-BUILD.bazel",
         strip_prefix = "pybind11-%s" % version,
-        urls = ["https://github.com/pybind/pybind11/archive/v%s.zip" % version],
+        url = "https://github.com/pybind/pybind11/archive/refs/tags/v%s.tar.gz" % version,
```

Comment:
Is this Github's preferred linking scheme? The releases page shows URLs like https://github.com/pybind/pybind11_bazel/releases/download/v2.12.0/pybind11_bazel-2.12.0.tar.gz

Is there any documentation for my own education?

Reply:
https://github.com/bazel-contrib/SIG-rules-authors/issues/11#issuecomment-1029861300

However your releases/download URL version is even better because those artifacts are _not_ generated by GitHub on demand but actual files attached to the release

Reply:
ah, it looks like there aren't actually any artifacts of that form for pybind11, only the generated ones. So will keep as is.
![image](https://github.com/user-attachments/assets/070f8709-7b9d-40af-a012-13c536532493)

Reply:
The one you pointed at is this repo itself, which is doing a better job :)
![image](https://github.com/user-attachments/assets/d1c37eb0-df81-4c3f-9d23-33ce7ff3dd10)

Reply:
The "Source code (...)" ones are generated by GitHub, anything else is manually uploaded as part of the release.

Reply:
Sorry my day job kept me very busy for a few days. This looks good, I'm going to merge it.